src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -634,6 +634,18 @@ public string DecryptToken(JsonWebToken jwtToken, TokenValidationParameters vali
                     continue;
                 }
 
+                if (validationParameters.AcceptedAlgorithms != null && validationParameters.AcceptedAlgorithms.Any() && !validationParameters.AcceptedAlgorithms.Contains(jwtToken.Enc, StringComparer.OrdinalIgnoreCase))
```

Comment:
we will need to add this to other call graphs: Saml1, Saml2, SignedHttpRequests, perhaps this logic should live in Validators.cs.

Reply:
@brentschmaltz Good point. This brings up a few questions when combined with changing the signature on the delegate

Should the SignedHttpRequestValidationParameters object also include restrictions on crypto algorithms? It would be confusing to reuse the TVP used for Access Token validation.

For SAML, there is not a SecurityToken available at the time that the signature is verified over the SignedInfo. Just a SecurityKey, TVP, and algorithm.

Is it better to broaden the inputs of the delegate with the trade off being that the logic will need to be able to handle potential null values?

Reply:
Following up, the saml token just needs to be brought into the context. and the SHR validation is out of scope for this task.